vllm/attention/backends/flashinfer.py

```diff
@@ -356,7 +357,8 @@ def build(self, seq_lens: List[int], query_lens: List[int],
             for i, block_table in enumerate(self.block_tables):
                 if block_table:
                     input_block_tables[i, :len(block_table)] = block_table
-            block_tables = torch.tensor(input_block_tables, device=device)
+            block_tables = torch.tensor(input_block_tables).to(
+                device, non_blocking=True)
```

Comment:
does this one actually work differently than before?

Reply:
I suppose to because `input_block_tables` is a numpy array, so `torch.tensor` needs to transfer the tensor to GPU anyways, so it was equivalent to `torch.tensor(input_block_tables).to(device)`. Now we make this transfer non-blocking so the transfer latency is supposed to be hidden by following Python-level operations.

Reply:
If `input_block_tables` is a numpy array, why not use `torch.from_numpy` to avoid copy?

Reply:
I don't know the rationale as it was already `torch.tensor`, but you're right we should use `from_numpy`.

Reply:
`non_blocking` only has effect if the CPU tensor is pinned, which it will not be in this case.

Reply:
Oh yeah good point, so this non-blocking should have no effect. The only benefit may just come from zero-copy by `from_numpy`.

Reply:
I think `.to(device, non_blocking=True)` should use [cudaMemcpyAsync](https://docs.nvidia.com/cuda/cuda-runtime-api/group__CUDART__MEMORY.html#group__CUDART__MEMORY_1g85073372f776b4c4d5f89f7124b7bf79), which can be async even if the cpu tensor is not pinned.

Reply:
You can see the link in the PR description, if the CPU tensor is pagable (not pinned), it the transfer is still blocked.